src/main/ruby/truffleruby/core/posix.rb

```diff
@@ -425,11 +425,74 @@ def self.read_string_native(io, length)
     end
   end
 
-  def self.read_to_buffer_native(io, length)
+  def self.execute_posix_read(io, buffer, length)
+    fd = io.fileno
+    return Truffle::POSIX.read(fd, buffer, length) unless io.timeout
+
+    deadline = Process.clock_gettime(Process::CLOCK_MONOTONIC) + io.timeout
+
+    while true do
```

Comment:
```suggestion
    while true
```
And we should add a RuboCop rule to catch this easily, maybe you can find it?

Reply:
Added the rule.